### ISSUE TITLE
Add warning about encodings not supported by Python

### ISSUE THREAD

**Issue by dan-blanchard** [chardet/chardet#71]
We currently detect `EUC-TW` pretty well, but it's [not actually supported by Python](http://bugs.python.org/issue2066).  Most users would expect that

``` python
result = chardet.detect(some_bytes)
try:
    some_bytes.decode(result['encoding'])
except UnicodeDecodeError:
    print('Oops. chardet detected the wrong encoding')
```

would always work, but the decode line can actually fail with a `LookupError` too because of encodings that aren't supported by Python.

**Comment by dan-blanchard** [comment]
I'm also considering whether or not we should even detect these legacy encodings by default.

**Comment by sigmavirus24** [comment]
If we want to deprecate those, we should save that removal until a major version bump and communicate it well.

**Comment by dan-blanchard** [comment]
Yeah. And I don't think we should remove them entirely. We should just not check for them by default.

**Comment by sigmavirus24** [comment]
So in that case, I'd like to propose a different API than what you might be thinking about for these issues.

I think a `Configuration` object that works like this might be good:

```py
from chardet import config

cfg = config.Configuration().enable(
    'EUC-TW'
    ).prerefine(
    'GB2312', 'Shift_JIS', 'EUC-TW',
    )
```

Then this would make the selections immutable and this could be passed like so:

```py
import chardet

chardet.detect(input_bytes, config=cfg)
```

**Comment by dan-blanchard** [comment]
Could you layout the advantages of that over just adding kwargs to `chardet.detect`?  

From my perspective I could see a configuration object being really handy if you were making loads of calls to `chardet.detect` at various points in your code, but my intuition is that most people are just calling it once (or repeatedly in a loop).

**Comment by sigmavirus24** [comment]
So, for one, I've grown to hate kwargs. Since we're supporting Python 2, it means we cannot make them keyword-only arguments which means that people *will* call them positionally. This makes it hard to deprecate features as we need to. Configuration objects as described above mean that we can make a method raise a DeprecationWarning, and removing it is as easy as making the method a no-op.

**Comment by dan-blanchard** [comment]
That's a really good point about deprecation.  I also hate people calling keyword arguments positionally, so I think you've got the right approach there.
